salt/data-pipeline/bigquery-views.sls

```diff
@@ -0,0 +1,39 @@
+bigquery-view docker image:
+    docker_image.present:
+        # without revision, 'latest' is assumed:
+        # https://docs.saltstack.com/en/latest/ref/states/all/salt.states.dockerng.html#salt.states.dockerng.image_present
+        - load: elifesciences/data-pipeline-bigquery-views
+        #- unless:
+        #    - test -d /vagrant
+
+clone biquery-views repo:
+    builder.git_latest:
+        - name: git@github.com:elifesciences/data-pipeline-bigquery-views
+        - identity: {{ pillar.elife.projects_builder.key or '' }}
+        - rev: master
+        - branch: master
+        - target: /opt/data-pipeline-bigquery-views
+        - force_fetch: True
+        - force_checkout: True
+        - force_reset: True
+
+    file.directory:
+        - name: /opt/data-pipeline-bigquery-views
+        - user: {{ pillar.elife.deploy_user.username }}
+        - group: {{ pillar.elife.deploy_user.username }}
+        - recurse:
+            - user
+            - group
+        - require:
+            - builder: clone biquery-views repo
+
+re-materialise views daily:
+    cron.present:
+        - user: {{ pillar.elife.deploy_user.username }}
+        - name: cd /opt/data-pipeline-bigquery-views && ./docker-views-cli.sh --dataset={{ pillar.elife.env }} materialize
```

Comment:
The bigquery-views repo does indeed have no dependency on a postgres database. I don't see postgres in this change?

You are also right that we don't need to check out the repo because the `views` directory (with `.sql` and `.lst` files) is copied to the container.

*Note*: one potential alternative to this approach. If we moved the `views-cli` code to a separate repo then that could be public and we could run that directly - in that case we would have to check out the repo though and mount the `views` directory (depends on https://github.com/elifesciences/data-pipeline-bigquery-views/issues/10). But we could continue to build and push a private image with the `views-cli` and `views`.

Reply:
> The bigquery-views repo does indeed have no dependency on a postgres database. I don't see postgres in this change?

Was a random example of how we should be able to use just the image of $software, rather than also its source

> If we moved the views-cli code to a separate repo then that could be public and we could run that directly

As long as it ends up in the image we shouldn't need anything more? It's more of a single-purpose Docker image rather than being very reusable, but at least it's a universal way to ship software where it's needed.

Reply:
The public views-cli, as a separate tool would not include the views itself. It then becomes a tool like `gcloud` or `terrarform` which one would call to deploy from config (here views) sourced from somewhere else. It could save us one private repo. But still having the private Docker image bundling both (views-cli and views) together would keep the formula slightly simpler and ensures that the we use the correct views-cli version together with the views.

Reply:
todo: add --rm
todo later: add image tag